src/main/java/knight/CommandParser.java

```diff
@@ -0,0 +1,74 @@
+package knight;
+
+/**
+ * Parses the user input to determine the command.
+ */
+public class CommandParser {
+    /**
+     * Parses the user input to determine the command.
+     *
+     * @param s The user input.
+     * @return The command type.
+     * @throws NonstandardCommandException If the command is not recognised.
+     */
+    static Command parseCommand(String s) {
+        if (s.equals("bye")) {
+            return Command.BYE;
+        } else if (s.equals("save")) {
+            return Command.SAVE;
+        } else if (s.equals("list")) {
+            return Command.LIST;
+        } else if (s.matches("mark [1-9]\\d*")) {
+            return Command.MARK;
+        } else if (s.matches("unmark [1-9]\\d*")) {
+            return Command.UNMARK;
+        } else if (s.matches("delete [1-9]\\d*")) {
+            return Command.DELETE;
+        } else if (s.matches("todo \\S.*")) {
+            return Command.TODO;
+        } else if (s.matches("deadline \\S.* /by \\S.*")) {
+            return Command.DEADLINE;
+        } else if (s.matches("event \\S.* /from \\S.* /to \\S.*")) {
+            return Command.EVENT;
+        } else if (s.matches("find \\S.*")) {
+            return Command.FIND;
```

Comment:
Cool use of regex to parse commands! Didn't think of that for mine

Reply:
yes agree, well done on using regex for each command!